fenix/app/src/main/java/org/mozilla/fenix/settings/logins/fragment/AddLoginFragment.kt

```diff
@@ -339,6 +339,15 @@ class AddLoginFragment : Fragment(R.layout.fragment_add_login), MenuProvider {
         activity?.invalidateMenu()
     }
 
+    private fun webProtocolTreatment(hostname: String): String {
```

Comment:
Can this check be moved to URLStringUtils.kt, where we can have something like:
```  
    /**
     * Returns true if the [String] uses the "http" or "https" protocol scheme.
     */
    fun String.isHttpOrHttps() =
        startsWith(HTTPS, ignoreCase = true) || startsWith(HTTP, ignoreCase = true)
```

then `webProtocolTreatment` function and companion object could be removed and even used directly as something like e.g.
```
interactor.onAddLogin(
                with(binding.hostnameText.text.toString()) {
                    if (isHttpOrHttps()) {
                        this
                    } else {
                        "$HTTPS_PROTOCOL$this"
                    }
                },
                binding.usernameText.text.toString(),
                binding.passwordText.text.toString(),
            )
```

You could then write tests for your `isHttpOrHttps` fun

Reply:
Refactored code and added unit tests.